src/AllContext/LoginContext.js

```diff
@@ -0,0 +1,29 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
```

Comment:
I have a suggestion, as this provider is not just login, it provides user details as well, so you should change loginContext to something more generic like AuthenticationContext